### PR TITLE
parse  synce logs to create metrics and events 

### PR DESCRIPTION
**Pull request by aneeshkp**
This PR will parse synce4l and create metrics to identify 
1. networkOptions
2. Clock Quality  metrics with indvidual ext-QL and  QL metrics 
3. Synce State metrics 

synce4l[622796.479]: [synce4l.0.config] tx_rebuild_tlv: attached new TLV, QL=0xf on ens7f0",
synce4l[622796.479]: [synce4l.0.config] tx_rebuild_tlv: attached new extended TLV, EXT_QL=0xff on ens7f0
synce4l[627602.540]: [synce4l.0.config]EEC_LOCKED/EEC_LOCKED_HO_ACQ on GNSS of synce1
synce4l[627602.540]: [synce4l.0.config] EEC_HOLDOVER on synce1
synce4l[627602.593]: [synce4l.0.config] tx_rebuild_tlv: attached new TLV, QL=0xf on ens7f0
synce4l[627602.593]: [synce4l.0.config] tx_rebuild_tlv: attached new extended TLV, EXT_QL=0xff on ens7f0
synce4l[627685.138]: [synce4l.0.config] EEC_LOCKED/EEC_LOCKED_HO_ACQ on GNSS of synce1
synce4l[627685.138]: [synce4l.0.config] act on EEC_LOCKED/EEC_LOCKED_HO_ACQ for ens7f0
Metrics 
` HELP openshift_ptp_synce_ssm_ql network_option1: ePRTC: {0, 0x2, 0x21}, PRTC:  {1, 0x2, 0x20}, PRC:   {2, 0x2, 0xFF}, SSUA:  {3, 0x4, 0xFF}, SSUB:  {4, 0x8, 0xFF}, EEC1:  {5, 0xB, 0xFF},\n    network_option2 ePRTC: {0, 0x1, 0x21}, PRTC:  {1, 0x1, 0x20}, PRS:   {2, 0x1, 0xFF}, STU:   {3, 0x0, 0xFF}, ST2:   {4, 0x7, 0xFF}, TNC:   {5, 0x4, 0xFF}, ST3E:  {6, 0xD, 0xFF}, EEC2:  {7, 0xA, 0xFF}, PROV:  {8, 0xE, 0xFF}
# TYPE openshift_ptp_synce_ssm_ql gauge
openshift_ptp_synce_ssm_ql{device="synce1",iface="ens7f0",network_option="2",node="cnfdg33.ptp.eng.rdu2.dc.redhat.com",process="synce4l",profile="synce4l.0.config",ql_type="Extended SSM"} 255
openshift_ptp_synce_ssm_ql{device="synce1",iface="ens7f0",network_option="2",node="cnfdg33.ptp.eng.rdu2.dc.redhat.com",process="synce4l",profile="synce4l.0.config",ql_type="SSM"} 1
openshift_ptp_synce_ssm_ql{device="synce1",iface="ens7f1",network_option="2",node="cnfdg33.ptp.eng.rdu2.dc.redhat.com",process="synce4l",profile="synce4l.0.config",ql_type="Extended SSM"} 255
openshift_ptp_synce_ssm_ql{device="synce1",iface="ens7f1",network_option="2",node="cnfdg33.ptp.eng.rdu2.dc.redhat.com",process="synce4l",profile="synce4l.0.config",ql_type="SSM"} 1`

### CODE REVIEW

**Review by vitus133 — COMMENTED** [review]
Added a couple of remarks, otherwise looks good!

File: pkg/daemon/config.go [hunk 1]
```diff
@@ -213,62 +194,73 @@ func (r *synceRelations) appendDevice(ifaces []string, devName string) {
...
+			}
+			if extendedTlvStr, ok := section.options["extended_tlv"]; ok {
+				extendedTlv, _ = strconv.Atoi(strings.TrimSpace(extendedTlvStr))
+			}
+			synceRelationInfo.NetworkOption = networkOption
+			synceRelationInfo.ExtendedTlv = extendedTlv
+		} else if strings.HasPrefix(section.sectionName, "[{") {
+			synceRelationInfo.ExternalSource = re.ReplaceAllString(section.sectionName, "")
+		} else if strings.HasPrefix(section.sectionName, "[") && section.sectionName != "[global]" &&
+			!strings.HasPrefix(section.sectionName, "[{") {
```
[the remark this answers is not in the document]
Comment (aneeshkp) [reply pkg/daemon/config.go:239]: Fixed

File: pkg/daemon/config.go [hunk 2]
```diff
@@ -213,62 +194,73 @@ func (r *synceRelations) appendDevice(ifaces []string, devName string) {
...
+				NetworkOption:  1,
+				ExtendedTlv:    0,
+				ExternalSource: "",
+				LastQLState:    make(map[string]*synce.QualityLevelInfo),
+				LastClockState: "",
+			}
+			extendedTlv, networkOption = 0, 1
+			synceRelationInfo.Name = re.ReplaceAllString(section.sectionName, "")
+			if networkOptionStr, ok := section.options["network_option"]; ok {
+				networkOption, _ = strconv.Atoi(strings.TrimSpace(networkOptionStr))
```
[the remark this answers is not in the document]
Comment (aneeshkp) [reply pkg/daemon/config.go:229]: done

File: pkg/daemon/daemon.go [hunk 3]
```diff
@@ -909,7 +934,16 @@ func (p *ptpProcess) cmdRun(stdoutToSocket bool) {
 
 // for ts2phc along with processing metrics need to identify event
 func (p *ptpProcess) processPTPMetrics(output string) {
-	if p.name == ts2phcProcessName && (strings.Contains(output, NMEASourceDisabledIndicator) ||
+	if p.name == syncEProcessName {
+		configName := strings.Replace(strings.Replace(p.messageTag, "]", "", 1), "[", "", 1)
+		if configName != "" {
+			configName = strings.Split(configName, MessageTagSuffixSeperator)[0] // remove any suffix added to the configName
+		}
+		//TODO Add sycne parsing logic
```
[the remark this answers is not in the document]
Comment (aneeshkp) [reply pkg/daemon/daemon.go:942]: done

**Pull request merged**